### PR TITLE
preferences are fetched from the server

### PR DESCRIPTION
**Pull request by mineralsfree**

### CODE REVIEW

File: src/app/api/users/route.js [hunk 1]
```diff
@@ -5,7 +5,7 @@ import { NextResponse } from 'next/server';
 export async function GET() {
   try {
     await connectToDataBase();
-    const users = await User.find().populate('preferences');
```
Comment (sebazai) [inline src/app/api/users/route.js:8]: Changes the users from array to obj

File: src/components/Map/Map.jsx [hunk 2]
```diff
@@ -5,18 +5,28 @@ import "leaflet/dist/leaflet.css";
...
 import { renderToString } from "react-dom/server";
+import {useChatMutation} from "@/store/services/chatApi";
 
 const Map = (props) => {
   const { position, placesData: places } = props;
-
+  const icon = L.icon({iconUrl: "/images/marker-icon.png"});
+  const [preferences, setPreferences] = useState([]);
   const mapRef = useRef();
+  const [chat, { isLoading, isSuccess, data, error }] = useChatMutation();
```
Comment (sebazai) [inline src/components/Map/Map.jsx:21]: This Chat mutation calls the wrong endpoint. You could use the page.jsx `fetchPlaces` function that is passed to the Map component to fetch based on preference string.

Also, why not use this hook in the component that uses this hook, why pass the fetch function as a prop, when we want to render everything and if it takes time to fetch the preferences, we could use the states to render later the preferences buttons.

File: src/components/Map/SelectionOverlay.jsx [hunk 3]
```diff
@@ -1,10 +1,40 @@
 import {Box, Chip} from "@mui/material";
+import * as MuiIcons from '@mui/icons-material';
 
-export const SelectionOverlay = (props)=>{
+export const SelectionOverlay = (props) => {
   const {chips, location, chat} = props;
+
   const onClick = (chip) => {
-    chat({ messages:  [{ role: "user", content: chip }], location });
+    chat({messages: [{role: "user", content: chip.description}], location});
```
Comment (sebazai) [inline src/components/Map/SelectionOverlay.jsx:8]: Shouldn't this be async if using the mutation

File: src/components/Map/SelectionOverlay.jsx [hunk 4]
```diff
@@ -1,10 +1,40 @@
...
+          boxShadow: '0px 1px 3px rgba(0, 0, 0, 0.2)', // Soft shadow for depth
+          margin: '0 4px',
+          padding: '0 8px',            // Padding to make it more button-like
+          height: '32px',              // Standard chip height
+          fontWeight: 500,             // Slightly bold text
+          fontSize: '0.875rem',        // Subtle font size
+          '& .MuiChip-icon': {
+            marginLeft: '4px',         // Space between icon and text
+            color: '#000',             // Ensure icon color is black
+          }}}
```
Comment (sebazai) [inline src/components/Map/SelectionOverlay.jsx:32]: Add to style sheet a a class.

File: src/components/Map/Map.jsx [hunk 5]
```diff
@@ -75,6 +81,9 @@ const Map = (props) => {
           attribution='&copy; <a href="https://www.openstreetmap.org/copyright">OpenStreetMap</a> contributors'
           url="https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png"
         ></TileLayer>
+        {data && data.places.map((item) => {
+          return <ItemMarker item={item} key={item.address} icon={icon} />;
+        })}
```
Comment (sebazai) [inline src/components/Map/Map.jsx:86]: If this now works, we can remove the useEffect in this component that renders the places.

File: src/components/Map/Map.jsx [hunk 6]
```diff
@@ -5,18 +5,28 @@ import "leaflet/dist/leaflet.css";
...
   const mapRef = useRef();
+  const [chat, { isLoading, isSuccess, data, error }] = useChatMutation();
 
+  useEffect(() => {
+    fetch('/api/users')
+      .then(response => response.json())
+      .then((data)=>{
+        setPreferences(data.preferences)
+      })
+  }, []);
```
Comment (sebazai) [inline src/components/Map/Map.jsx:29]: Would move to page.jsx that renders the component and pass as props.

**Review by sebazai — CHANGES_REQUESTED** [review]
Also getting unhandled runtime errors.

File: src/components/Map/Map.jsx (hunk 6 above)
Comment (sebazai) [reply src/components/Map/Map.jsx:29]:
```
{
error: "Schema hasn't been registered for model "Preference".
Use mongoose.model(name, schema)"
}
```

**Comment by mineralsfree** [comment]
> Also getting unhandled runtime errors.

Please, provide more details

File: src/app/api/users/route.js (hunk 1 above)
Comment (mineralsfree) [reply src/app/api/users/route.js:8]: It just fetch one user instead of  all users, exactly  what we need

File: src/components/Map/Map.jsx (hunk 5 above)
Comment (mineralsfree) [reply src/components/Map/Map.jsx:86]: Does it work for you? If yes, we can refactor it later. We need features now

File: src/components/Map/SelectionOverlay.jsx (hunk 3 above)
Comment (mineralsfree) [reply src/components/Map/SelectionOverlay.jsx:8]: it just dispatches an action, no need for async / .then

File: src/app/api/users/route.js (hunk 1 above)
Comment (sebazai) [reply src/app/api/users/route.js:8]: True, and I guess this was not used anywhere before, so does not matter. I receive 500 error, I guess it's related to this I commented below: 
{
error: "Schema hasn't been registered for model "Preference".
Use mongoose.model(name, schema)"
}

**Comment by sebazai** [comment]
> git pull git@github.com:mineralsfree/TravelMinds.git fetch-preferences

```
Unhandled Runtime Error
TypeError: Cannot read properties of undefined (reading 'map')

Source
src/components/Map/SelectionOverlay.jsx (10:33) @ map

   8 |     chat({messages: [{role: "user", content: chip.description}], location});
   9 |   }
> 10 |   const chipsComponents = chips.map((chip, index) => {
     |                                 ^
  11 |     const IconComponent = MuiIcons[chip.icon];
  12 |     return (<Chip
  13 |       icon={IconComponent ? <IconComponent color={"#000"}/> : null}
  ```

File: src/components/Map/Map.jsx (hunk 5 above)
Comment (sebazai) [reply src/components/Map/Map.jsx:86]: I do not know, as I receive runtime errors before it renders the itemmarkers.

**Comment by mineralsfree** [comment]
> > git pull [git@github.com](mailto:git@github.com):mineralsfree/TravelMinds.git fetch-preferences
> 
> ```
> Unhandled Runtime Error
> TypeError: Cannot read properties of undefined (reading 'map')
> 
> Source
> src/components/Map/SelectionOverlay.jsx (10:33) @ map
> 
>    8 |     chat({messages: [{role: "user", content: chip.description}], location});
>    9 |   }
> > 10 |   const chipsComponents = chips.map((chip, index) => {
>      |                                 ^
>   11 |     const IconComponent = MuiIcons[chip.icon];
>   12 |     return (<Chip
>   13 |       icon={IconComponent ? <IconComponent color={"#000"}/> : null}
> ```

I've updated code that might fix the issue, please, retest :)

**Comment by sebazai** [comment]
> git pull git@github.com:mineralsfree/TravelMinds.git fetch-preferences

Hmm, now it seems it can't find @mui/styles package, did you forget to install it?

**Comment by mineralsfree** [comment]
> > git pull [git@github.com](mailto:git@github.com):mineralsfree/TravelMinds.git fetch-preferences
> 
> Hmm, now it seems it can't find @mui/styles package, did you forget to install it?

Forgot to include package.json in commit, sorry :face_exhaling:

**Review by sebazai — APPROVED** [review]

**Pull request merged**